include/aws/http/private/http2_stream_manager_impl.h

```diff
@@ -151,6 +151,11 @@ struct aws_http2_stream_manager {
         size_t internal_refcount_stats[AWS_SMCT_COUNT];
 
         bool finish_pending_stream_acquisitions_task_scheduled;
+
+        /**
+         * Number of streams can be made without make any new connections. Only used for metric purpose.
+         */
+        size_t available_concurrency;
```

Comment:
naming: I like `metric_available_concurrency`. helps emphasize that we're not using this to drive behavior